File: platforms/web/examples/core/renderer/Renderer.js

import { Shader, Renderer } from "fragmentcolor";

const renderer = new Renderer();

// Use your platform's windowing system to create a window
const canvas = document.createElement('canvas');

// Create a Target from it
const target = await renderer.createTarget(canvas);
const texture_target = await renderer.createTextureTarget([16, 16]);

// RENDERING
renderer.render(new Shader(""), texture_target);

// That's it. Welcome to FragmentColor!

File: platforms/kotlin/examples/core/renderer/Renderer.kt

import org.fragmentcolor.*

val renderer = Renderer()

// Use your platform's windowing system to create a window
// HEADLESS: canvas creation not needed on Android

// Create a Target from it
val target = renderer.createTextureTarget(800u, 600u)
val texture_target = renderer.createTextureTarget(16u, 16u)

// RENDERING
renderer.render(Shader(""), texture_target)

// That's it. Welcome to FragmentColor!